src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs

```diff
@@ -331,6 +342,144 @@ private void WritePagingOperation(CodeWriter writer, Paging pagingMethod, bool a
+
+            using (writer.Scope())
+            {
+                WriteParameterNullChecks(writer, parameters);
+
+                writer.Append($"return {typeof(ArmOperationHelpers)}.Create(");
+                writer.Line($"pipeline, clientDiagnostics, originalResponse, {typeof(RequestMethod)}.{originalMethod.Request.HttpMethod.ToRequestMethodName()}, {originalMethod.Diagnostics.ScopeName:L}, {typeof(FinalStateVia)}.{lroMethod.FinalStateVia}, createOriginalHttpMessage,");
+
+                string valueVariable = "value";
+                const string document = "document";
+                ObjectSerialization? serialization = (lroMethod.OriginalResponse.ResponseBody as ObjectResponseBody)?.Serialization;
+                writer.Line($"(r, c) =>");
+                writer.LineRaw("{");
+                switch (serialization)
+                {
+                    case JsonSerialization jsonSerialization:
+                        writer.Append($"using var {document:D} = ");
+                        writer.Line($"{typeof(JsonDocument)}.Parse(r.ContentStream);");
+                        writer.ToDeserializeCall(
+                            jsonSerialization,
+                            w => w.Append($"document.RootElement"),
+                            ref valueVariable
+                        );
+                        writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, r);");
+                        break;
+                    case XmlElementSerialization xmlSerialization:
+                        writer.Line($"var {document:D} = {typeof(XDocument)}.Load(r.ContentStream, LoadOptions.PreserveWhitespace);");
+                        writer.ToDeserializeCall(
+                            xmlSerialization,
+                            w => w.Append($"document"),
+                            ref valueVariable
+                        );
+                        writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, r);");
+                        break;
+                    default:
+                        writer.Line($"return Response.FromValue(r, r);");
+                        break;
+                }
+                writer.LineRaw("},");
+                writer.Line($"async (r, c) =>");
+                writer.LineRaw("{");
+                switch (serialization)
+                {
+                    case JsonSerialization jsonSerialization:
+                        writer.Append($"using var {document:D} = ");
+                        writer.Line($"await {typeof(JsonDocument)}.ParseAsync(r.ContentStream, default, c).ConfigureAwait(false);");
+                        writer.ToDeserializeCall(
+                            jsonSerialization,
+                            w => w.Append($"document.RootElement"),
+                            ref valueVariable
+                        );
+                        writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, r);");
```

Comment:
Can we create `Response<T>` in the LongRunningOperation implementation? It would simplify generated code a little bit more.

Reply:
This is our deserialization logic that is copy-pasted from how we write deserialization already. So, this is dynamic based on what type we're trying to deserialize into. Is there a more generic process we could use everywhere? If so, it would apply here too.

Reply:
The difference is that in other cases we don't have a hand-written piece of logic in between. Usually methods return Response<T> directly so there is nowhere we can put the dynamic logic. But here there is an obvious place for it.

Reply:
I'm confused as the code generated here calls generated code deserializers. So, unless we make interfaces for our deserializers, there's not an easy path to have a generic deserialization mechanism. If we want that, feel free to make an issue and we can address it in a follow-up PR, for both this situation and our overall deserialization writing.

Reply:
Fixed. Now returns just the Value.